eng/tools/suppressions/src/index.ts

```diff
@@ -0,0 +1,141 @@
+import { access, constants, readFile } from "fs/promises";
+import { minimatch } from "minimatch";
+import { dirname, join, resolve } from "path";
+import { exit } from "process";
+import { parse as yamlParse } from "yaml";
+import { z } from "zod";
+import { fromError } from "zod-validation-error";
+
+interface Suppression {
+  tool: string;
+  path: string;
+  reason: string;
+}
+
+const suppressionSchema = z.array(
+  z.object({
+    tool: z.string(),
+    path: z.string(),
+    reason: z.string(),
+  }),
+);
+
+export async function main() {
+  const args = process.argv.slice(2);
+
+  if (args.length === 2) {
+    const tool = args[0];
+    const path = args[1];
+    const suppressions = await getSuppressions(tool, path);
+    console.log(JSON.stringify(suppressions));
+    exit(0);
+  } else {
+    console.error(getUsage());
+    exit(1);
+  }
+}
+
+function getUsage(): string {
+  return (
+    "  Usage: npx get-suppressions <tool-name> <path-to-file>\n" +
+    "Returns: JSON array of suppressions, with specified tool name, applying to file (may be empty)\n" +
+    "\n" +
+    "Example: npx get-suppressions TypeSpecRequirement specification/foo/data-plane/Foo/stable/2023-01-01/Foo.json\n" +
+    'Returns: [{"tool":"TypeSpecRequirement","path":"**/*.json","reason":"foo"}]\n'
+  );
+}
+
+// tool: Name of tool
```

Comment:
```suggestion
// tool: Name of tool. It is matched against the values of `tool` key in the suppressions file.
```

Reply:
if its a known set can also define that as an `z.enum(["foo", "bar"])` so it validates that

Reply:
I explicitly do **not** want to hardcode the list of tools in the shared component.  I want to keep this flexible so we can add tools without needing to update the core.

Reply:
If we start getting reports from users that are using tool names with typos, I might reconsider.  But for now I prefer the flexibility over the validation.